src/rr/dnssec/algorithm.rs

```diff
@@ -230,7 +310,15 @@ impl Algorithm {
         bytes.extend_from_slice(&n);
 
         bytes
-      }
+      },
+      Algorithm::ECDSAP256SHA256 | Algorithm::ECDSAP384SHA384 => {
+        let pkey = if let DnssecKey::Ecdsa(ref key) = *public_key {
+            key
+        } else {
+            panic!("Invalid type of key")
+        };
+        pkey.to_wire()
```

Comment:
Admittedly I haven't tested it yet, but I am pretty sure it is correct. Anyways, I will of course add test cases to ensure it actually works (after verification is implemented). Since [RFC 6605](https://tools.ietf.org/html/rfc6605) provides examples of working keys and signatures, it shouldn't be too hard to confirm. :)